tools/compiletest/src/main.rs

```diff
@@ -387,16 +387,24 @@ fn collect_tests_from_dir(
     // output directory corresponding to each to avoid race conditions during
     // the testing phase. We immediately return after adding the tests to avoid
     // treating `*.rs` files as tests.
-    if config.mode == Mode::CargoKani && dir.join("Cargo.toml").exists() {
+    if config.mode == Mode::CargoKani {
```

Comment:
This function has become rather awkward. I think I'm not opposed to this, but perhaps we should break this function into two? `collect_expected_tests_from_dir` and `collect_rs_tests_from_dir`?

Reply:
Good idea. Done.